### PR TITLE
Design Review for new DFW API

### PR DESCRIPTION
**Pull request by madhephaestus**
The users code is not isolated in seperate files and a pointer of DFW will be passed with the current time to the user call back lambda.

### CODE REVIEW

File: DFW.cpp [hunk 1]
```diff
@@ -136,172 +206,128 @@ int DFW::joysticklh(void) // returns joystick right vertical
 int DFW::start(void) // returns start button state
```
Comment (AustinShalit) [inline DFW.cpp]: All of the button functions should return bool (true meaning depressed, false otherwise).

File: DFW.cpp [hunk 2]
```diff
@@ -136,172 +206,128 @@ int DFW::joysticklh(void) // returns joystick right vertical
 int DFW::start(void) // returns start button state 
 		{
 	int startB;
```
Comment (AustinShalit) [inline DFW.cpp]: For all of these functions: Unused variable

File: DFW.h [hunk 3]
```diff
@@ -1,18 +1,33 @@
-// Serial controller library for DFWireless controllers
+/**
+ * Serial controller library for DFWireless controllers
```
Comment (AustinShalit) [inline DFW.h:2]: We might have scope creep with the DFW library.  Now instead of just handling the controller, it is also handling all of the competition stuff.

File: DFW.h [hunk 4]
```diff
@@ -1,18 +1,33 @@
-// Serial controller library for DFWireless controllers
+/**
+ * Serial controller library for DFWireless controllers
+ * For RBE Class use
+ * @author Joe St. Germain joest@wpi.edu
+ * @author Alex Taglieri (atags22)
+ * @author Kevin Harrington (madhephaestus)
```
Comment (AustinShalit) [inline DFW.h:6]: I do not think we should include author tags.  Just use the git history.

File: DFW.h [hunk 5]
```diff
@@ -1,18 +1,33 @@
...
+ */
 //written by Joe St. Germain
-//For RBE Class use
-//joest@wpi.edu
-//@author Alex Taglieri (atags22)
-
+//
+//
+//
```
Comment (AustinShalit) [inline DFW.h:11]: Remove these extra `//`

**Review by AustinShalit — CHANGES_REQUESTED** [review]
This is probably a move in the right direction for the library.  A few general notes:
• See my note about scope creep.
• There is no style guide right now.  I recommend using either the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) or the [WPILib Style Guide](https://github.com/wpilibsuite/styleguide)
• This does not support the pause after autonomous to count up the score.  This is a very important feature.
• Please squash all commits

File: DFW.cpp (hunk 1 above)
Comment (madhephaestus) [reply DFW.cpp]: i fixed this

File: DFW.cpp (hunk 2 above)
Comment (madhephaestus) [reply DFW.cpp]: i fixed this too

File: DFW.h (hunk 3 above)
Comment (madhephaestus) [reply DFW.h:2]: This i disagree with. This library is not just a controller reader, it is the competition framework. Explicatly it is a framework, not a library. I talked this over with Miller, Stafford and Gennert. This model is what they want to see.

File: DFW.h (hunk 4 above)
Comment (madhephaestus) [reply DFW.h:6]: Commit history is good too, except Joes contributions would be obliterated. I want to keep the author tags wherever possible.

**Comment by madhephaestus** [comment]
commits are squashed on the merge step in GitHub, btw...
